airbyte-integrations/connectors/source-postgres/src/test-integration/java/io/airbyte/integrations/io/airbyte/integration_tests/sources/AbstractPostgresSourceDatatypeTest.java

```diff
@@ -0,0 +1,528 @@
+            .addInsertValues("null", "decode('1234', 'hex')", "'1234'", "'abcd'", "'\\xabcd'")
+            .addExpectedValues(null, "\\x1234", "\\x31323334", "\\x61626364", "\\xabcd")
+            .build());
+
+    for (final String type : Set.of("character", "char")) {
+      addDataTypeTestData(
+          TestDataHolder.builder()
+              .sourceType(type)
+              .airbyteType(JsonSchemaType.STRING)
+              .addInsertValues("'a'", "'*'", "null")
+              .addExpectedValues("a", "*", null)
+              .build());
+
+      addDataTypeTestData(
+          TestDataHolder.builder()
+              .sourceType(type)
+              .fullSourceDataType(type + "(8)")
+              .airbyteType(JsonSchemaType.STRING)
+              .addInsertValues("'{asb123}'", "'{asb12}'")
+              .addExpectedValues("{asb123}", "{asb12} ")
+              .build());
+    }
+
+    for (final String type : Set.of("varchar", "text")) {
+      addDataTypeTestData(
+          TestDataHolder.builder()
+              .sourceType(type)
+              .airbyteType(JsonSchemaType.STRING)
+              .addInsertValues("'a'", "'abc'", "'Миші йдуть на південь, не питай чому;'", "'櫻花分店'",
+                  "''", "null", "'\\xF0\\x9F\\x9A\\x80'")
+              .addExpectedValues("a", "abc", "Миші йдуть на південь, не питай чому;", "櫻花分店", "",
+                  null, "\\xF0\\x9F\\x9A\\x80")
+              .build());
+    }
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("varchar")
+            .fullSourceDataType("character varying(10)")
+            .airbyteType(JsonSchemaType.STRING)
+            .addInsertValues("'{asb123}'", "'{asb12}'")
+            .addExpectedValues("{asb123}", "{asb12}")
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("cidr")
+            .airbyteType(JsonSchemaType.STRING)
+            .addInsertValues("null", "'192.168.100.128/25'", "'192.168/24'", "'192.168.1'",
+                "'128.1'", "'2001:4f8:3:ba::/64'")
+            .addExpectedValues(null, "192.168.100.128/25", "192.168.0.0/24", "192.168.1.0/24",
+                "128.1.0.0/16", "2001:4f8:3:ba::/64")
+            .build());
+
+    addDataTypeTestData(
+        TestDataHolder.builder()
+            .sourceType("circle")
+            .airbyteType(JsonSchemaType.STRING)
+            .addInsertValues("'(5,7),10'", "'(0,0),0'", "'(-10,-4),10'", "null")
+            .addExpectedValues("<(5.0,7.0),10.0>", "<(0.0,0.0),0.0>", "<(-10.0,-4.0),10.0>", null)
+            .build());
+
+    // Debezium does not handle era indicators (AD nd BC)
+    // https://github.com/airbytehq/airbyte/issues/14590
+     addDataTypeTestData(
+     TestDataHolder.builder()
+     .sourceType("date")
+     .airbyteType(JsonSchemaType.STRING_DATE)
+     .addInsertValues("'1999-01-08'", /* "'1991-02-10 BC'",*/ "null")
+     .addExpectedValues("1999-01-08", /* "1990-02-10 BC", */ null)
```

Comment:
```suggestion
     .addExpectedValues("1999-01-08", /* "1991-02-10 BC", */ null)
```
(unless that's intentional?)